### ISSUE TITLE
同样的情况下，我用navicat可以连上mysql数据库，但是用Dbever连接不上。

### ISSUE THREAD

**Issue by kasiry** [dbeaver/dbeaver#17342]
### Description

mysql数据库部署在远程Liunx服务器上，网络有限制需要连接VPN；在连接VPN的情况下，我是用navicat可以连上该数据库，我使用dbever连不上；我又换了种测试方式，我不使用VPN，使用和Liunx服务器同网段的网络也可以连上数据库，请问下，这种情况应该配置Dbever的哪里呢？

### DBeaver Version

21.3.2

### Operating System

windows10

### Database and driver

Mysql数据8.X  驱动：com.mysql.cj.jdbc.Driver 8.0.17

### Steps to reproduce

背景：
mysql数据库部署在客户现场的Liunx服务器上，有两种工作人员，一种是在现场的工作人员，一种是在远程的工作人员。
1、在现场的工作使用和服务器相同的网络即客户内部网络，使用Dbever可以连接上数据库。
2、在远程的工作人员需要先连接客户方的VPN，再连接数据库。但是远程工作人员可以使用navicat连接上数据库，无法使用Dbever连接上数据库。
3、现场工作人员，解除现场网络，使用手机热点，连接VPN同样可以使用navicat连接数据库，无法使用Dbever连接数据库。
综上：这种情况需要修改Dbever的什么配置呢？最好能截图看看，谢谢！

**Comment by LonwoLonwo** [comment]
Hi @kasiry 

Please translate your ticket into English.

**Comment by kasiry** [comment]
ok。
Description
The mysql database is deployed on the remote Liunx server, and VPN is required for network restrictions. In the case of VPN connection, I can connect to the database using Navicat, I can not connect to the database using Dbever; I changed the test method again, I do not use VPN, use the same network segment as Liunx server can also connect to the database, may I ask, in this case, where should I configure Dbever?

DBeaver Version
21.3.2

Operating System
windows10

Database and driver
database：mysql 8.x    driver：com.mysql.cj.jdbc.Driver 8.0.17

Steps to reproduce
Background:
The mysql database is deployed on the Liunx server at the customer site with two types of staff, one on site staff and one remote staff.
1. The work on site uses the same network as the server, namely the customer's internal network, and Dbever can be used to connect to the database.
2. Remote workers need to connect to the client's VPN first and then to the database. However, remote workers can connect to the database using Navicat, not Dbever.
3, the field staff, remove the field network, use the mobile phone hotspot, connect VPN can also use Navicat to connect to the database, can not use Dbever to connect to the database.
In summary: What configuration of Dbever needs to be modified in this case? It's better to take a screenshot, thank you!

**Comment by kasiry** [comment]
ok。
Description
The mysql database is deployed on the remote Liunx server, and VPN is required for network restrictions. In the case of VPN connection, I can connect to the database using Navicat, I can not connect to the database using Dbever; I changed the test method again, I do not use VPN, use the same network segment as Liunx server can also connect to the database, may I ask, in this case, where should I configure Dbever?

DBeaver Version
21.3.2

Operating System
windows10

Database and driver
database：mysql 8.x    driver：com.mysql.cj.jdbc.Driver 8.0.17

Steps to reproduce
Background:
The mysql database is deployed on the Liunx server at the customer site with two types of staff, one on site staff and one remote staff.
1. The work on site uses the same network as the server, namely the customer's internal network, and Dbever can be used to connect to the database.
2. Remote workers need to connect to the client's VPN first and then to the database. However, remote workers can connect to the database using Navicat, not Dbever.
3, the field staff, remove the field network, use the mobile phone hotspot, connect VPN can also use Navicat to connect to the database, can not use Dbever to connect to the database.
In summary: What configuration of Dbever needs to be modified in this case? It's better to take a screenshot, thank you!



> Hi @kasiry
> 
> Please translate your ticket into English.

ok。
Description
The mysql database is deployed on the remote Liunx server, and VPN is required for network restrictions. In the case of VPN connection, I can connect to the database using Navicat, I can not connect to the database using Dbever; I changed the test method again, I do not use VPN, use the same network segment as Liunx server can also connect to the database, may I ask, in this case, where should I configure Dbever?

DBeaver Version
21.3.2

Operating System
windows10

Database and driver
database：mysql 8.x    driver：com.mysql.cj.jdbc.Driver 8.0.17

Steps to reproduce
Background:
The mysql database is deployed on the Liunx server at the customer site with two types of staff, one on site staff and one remote staff.
1. The work on site uses the same network as the server, namely the customer's internal network, and Dbever can be used to connect to the database.
2. Remote workers need to connect to the client's VPN first and then to the database. However, remote workers can connect to the database using Navicat, not Dbever.
3, the field staff, remove the field network, use the mobile phone hotspot, connect VPN can also use Navicat to connect to the database, can not use Dbever to connect to the database.
In summary: What configuration of Dbever needs to be modified in this case? It's better to take a screenshot, thank you!

**Comment by kasiry** [comment]
> Hi @kasiry
> 
> Please translate your ticket into English.
ok。
Description
The mysql database is deployed on the remote Liunx server, and VPN is required for network restrictions. In the case of VPN connection, I can connect to the database using Navicat, I can not connect to the database using Dbever; I changed the test method again, I do not use VPN, use the same network segment as Liunx server can also connect to the database, may I ask, in this case, where should I configure Dbever?

DBeaver Version
21.3.2

Operating System
windows10

Database and driver
database：mysql 8.x    driver：com.mysql.cj.jdbc.Driver 8.0.17

Steps to reproduce
Background:
The mysql database is deployed on the Liunx server at the customer site with two types of staff, one on site staff and one remote staff.
1. The work on site uses the same network as the server, namely the customer's internal network, and Dbever can be used to connect to the database.
2. Remote workers need to connect to the client's VPN first and then to the database. However, remote workers can connect to the database using Navicat, not Dbever.
3, the field staff, remove the field network, use the mobile phone hotspot, connect VPN can also use Navicat to connect to the database, can not use Dbever to connect to the database.
In summary: What configuration of Dbever needs to be modified in this case? It's better to take a screenshot, thank you!

**Comment by kasiry** [comment]
> Hi @kasiry
> 
> Please translate your ticket into English.


OK。

Description
The mysql database is deployed on the remote Liunx server, and VPN is required for network restrictions. In the case of VPN connection, I can connect to the database using Navicat, I can not connect to the database using Dbever; I changed the test method again, I do not use VPN, use the same network segment as Liunx server can also connect to the database, may I ask, in this case, where should I configure Dbever?

DBeaver Version
21.3.2

Operating System
windows10

Database and driver
database：mysql 8.x    driver：com.mysql.cj.jdbc.Driver 8.0.17

Steps to reproduce
Background:
The mysql database is deployed on the Liunx server at the customer site with two types of staff, one on site staff and one remote staff.
1. The work on site uses the same network as the server, namely the customer's internal network, and Dbever can be used to connect to the database.
2. Remote workers need to connect to the client's VPN first and then to the database. However, remote workers can connect to the database using Navicat, not Dbever.
3, the field staff, remove the field network, use the mobile phone hotspot, connect VPN can also use Navicat to connect to the database, can not use Dbever to connect to the database.
In summary: What configuration of Dbever needs to be modified in this case? It's better to take a screenshot, thank you!

**Comment by birdingman0626** [comment]
用IP地址连，不要用hostname

**Comment by uslss** [comment]
there has been no update on the issue for a long time. If it is still actual - feel free to reopen it

**Comment by StarryH2** [comment]
目前我下载的新版本也是有这个问题，请问如何解决呢？
The new version I have downloaded currently also has this issue. How can I solve it?

**Comment by StarryH2** [comment]
补充一下，刚刚倒找解决方式了。
在dbeaver/dbeaver.ini 文件下加上
-Djava.net.preferIPv4Stack=true

**Comment by LonwoLonwo** [comment]
Please create a separate ticket with all the details in English. We will look at it.
